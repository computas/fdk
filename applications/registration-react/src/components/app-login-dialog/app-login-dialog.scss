@import '../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.login-dialog-wrapper {
  background-color: $color-cta4;
  flex-grow: 1;
  .fdk-button {
    text-decoration: none;
    &:hover {
      color: $color0;
    }
  }
}

/*==========  Media Queries  ==========*/
/* Custom, iPhone Retina */
@media only screen and (min-width : 320px) {

}/* end min-width : 320px */


/* Extra Small Devices, Phones */
@media only screen and (min-width : 480px) {

}/* end min-width : 480px */

/* Small Devices, Tablets */
@media only screen and (min-width: 768px) {
  .login-dialog-wrapper {
    background-image: url(../../assets/img/illustration-login.svg);
    background-repeat: no-repeat;
    background-position: top right;
    background-size: 50%;
  }
}

/* Medium Devices, Desktops */
@media only screen and (min-width : 992px) {
  .login-dialog-wrapper {
    background-size: contain;
  }
}

/* Large Devices, Wide Screens */
@media only screen and (min-width : 1200px) {
}

