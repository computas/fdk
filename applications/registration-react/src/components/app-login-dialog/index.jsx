import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import axios from 'axios';

import { resetUser } from '../../actions/index';
import localization from '../../utils/localization';
import './index.scss';

// const LoginDialog = props => {
export class LoginDialog extends React.Component {
  componentWillMount() {
    if (this.props.loggedOut) {
      axios
        .get('/logout')
        .then(() => {
          this.props.dispatch(resetUser());
        })
        .catch(() => {
          this.props.dispatch(resetUser());
        });
    }
  }
  render() {
    const { loggedOut } = this.props;
    return (
      <div className="login-dialog-wrapper p-5">
        <div className="container">
          <div className="row">
            <div className="col-12">
              <div className="col-12 col-md-8">
                {loggedOut && (
                  <div
                    className="mt-2 alert alert-warning fdk-text-size-small fdk-color1"
                    role="alert"
                  >
                    <span>
                      <strong>
                        {localization.loginDialog.loggedOutMsgPart1}
                      </strong>
                    </span>
                    <span>{localization.loginDialog.loggedOutMsgPart2}</span>
                  </div>
                )}
                {!loggedOut && (
                  <div>
                    <h1 className="fdk-text-extra-strong mb-md-5">
                      {localization.loginDialog.title}
                    </h1>
                    <div className="fdk-text-size-medium fdk-text-line-medium mt-2 mb-md-3">
                      {localization.loginDialog.ingress}
                    </div>
                  </div>
                )}
                <div className="mt-5 mb-5">
                  <a className="fdk-button fdk-button-cta" href="/login">
                    {localization.app.logIn}
                  </a>
                </div>
                <div className="fdk-text-size-small fdk-text-line-medium">
                  <strong>
                    {localization.catalogs.missingCatalogs.accessTitle}
                  </strong>
                  <p>{localization.catalogs.missingCatalogs.accessText}</p>
                  <strong>
                    {localization.catalogs.missingCatalogs.assignAccessTitle}
                  </strong>
                  <p>
                    {localization.catalogs.missingCatalogs.assignAccessText}
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

LoginDialog.defaultProps = {
  loggedOut: false
};

LoginDialog.propTypes = {
  dispatch: PropTypes.func.isRequired,
  loggedOut: PropTypes.bool
};

function mapStateToProps({ user }) {
  const { userItem, isFetchingUser } = user || {
    userItem: null,
    isFetchingUser: false
  };
  return {
    userItem,
    isFetchingUser
  };
}

export default connect(mapStateToProps)(LoginDialog);
