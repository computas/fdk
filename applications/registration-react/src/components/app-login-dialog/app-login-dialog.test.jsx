import React from 'react';
import { shallow } from 'enzyme';
import { LoginDialog } from './app-login-dialog.component';

let defaultProps;
let wrapper;
let dispatch;

beforeEach(() => {
  dispatch = jest.fn();
  defaultProps = {
    dispatch,
    loggedOut: true
  };
  wrapper = shallow(<LoginDialog {...defaultProps} />);
});

test('should render LoginDialog correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
