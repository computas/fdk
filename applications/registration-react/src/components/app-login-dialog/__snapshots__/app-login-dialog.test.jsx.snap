// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginDialog correctly 1`] = `
<div
  className="login-dialog-wrapper p-5"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <div
          className="col-12 col-md-8"
        >
          <div
            className="mt-2 alert alert-warning fdk-text-size-small fdk-color1"
            role="alert"
          >
            <span>
              <strong>
                Du har blitt logget ut. 
              </strong>
            </span>
            <span>
              Sesjonen ble tidsavbrutt, men vi har tatt vare på alle endringene du har gjort. Logg inn på nytt for å fortsette å jobbe med registreringen.
            </span>
          </div>
          <div
            className="mt-5 mb-5"
          >
            <a
              className="fdk-button fdk-button-cta"
              href="/login"
            >
              Logg inn
            </a>
          </div>
          <div
            className="fdk-text-size-small fdk-text-line-medium"
          >
            <strong>
              Trenger du tilgang?
            </strong>
            <p>
              Kontakt din leder for å få tilgang til å registrere datasett.
            </p>
            <strong>
              Skal du gi andre tilgang?
            </strong>
            <p>
              Du kan gi andre tilgang til å redigere datasett for din virksomhet via Altinn. Les mer om hvordan du gir andre tilgang.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
